core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -520,86 +622,53 @@ public static Timer timer(String name) {
   /**
-   * @return all the worker's metrics in the format of {@link Metric}
-   */
-  public static List<Metric> allWorkerMetrics() {
-    return allInstanceMetrics(InstanceType.WORKER);
-  }
-
-  /**
-   * @return all the client's metrics in the format of {@link Metric}
-   */
-  public static List<Metric> allClientMetrics() {
-    return allInstanceMetrics(InstanceType.CLIENT);
-  }
-
-  /**
-   * Gets all metrics of the given instance type.
+   * Initialize the should report metrics. This should be called only once.
    *
-   * @param instanceType the requested instance type
-   * @return all metrics of the given instance type
-   */
-  private static List<Metric> allInstanceMetrics(MetricsSystem.InstanceType instanceType) {
-    List<Metric> metrics = new ArrayList<>();
-    for (Entry<String, Gauge> entry : METRIC_REGISTRY.getGauges().entrySet()) {
-      if (entry.getKey().startsWith(instanceType.toString())) {
-        Object value = entry.getValue().getValue();
-        if (!(value instanceof Number)) {
-          LOG.warn("The value of metric {} of type {} is not sent to metrics master,"
-                  + " only metrics value of number can be collected",
-              entry.getKey(), entry.getValue().getClass().getSimpleName());
-          continue;
-        }
-        metrics.add(Metric.from(entry.getKey(), ((Number) value).longValue(), MetricType.GAUGE));
-      }
-    }
-    for (Entry<String, Counter> entry : METRIC_REGISTRY.getCounters().entrySet()) {
-      if (entry.getKey().startsWith(instanceType.toString())) {
-        metrics.add(Metric.from(entry.getKey(), entry.getValue().getCount(), MetricType.COUNTER));
-      }
-    }
-    for (Entry<String, Meter> entry : METRIC_REGISTRY.getMeters().entrySet()) {
-      if (entry.getKey().startsWith(instanceType.toString())) {
-        // TODO(yupeng): From Meter's implementation, getOneMinuteRate can only report at rate of at
-        // least seconds. if the client's duration is too short (i.e. < 1s), then getOneMinuteRate
-        // would return 0
-        metrics.add(Metric.from(entry.getKey(), entry.getValue().getOneMinuteRate(),
-            MetricType.METER));
-      }
-    }
-    for (Entry<String, Timer> entry : METRIC_REGISTRY.getTimers().entrySet()) {
-      if (entry.getKey().startsWith(instanceType.toString())) {
-        metrics.add(Metric.from(entry.getKey(), entry.getValue().getCount(), MetricType.TIMER));
-      }
+   * Note that this method is able to catch most of the should report metrics
+   * except worker metrics with ufs tags.
+   *
+   * @param instanceType the instance type
+   */
+  @VisibleForTesting
+  public static void initShouldReportMetrics(InstanceType instanceType) {
+    Set<MetricKey> metricKeys = MetricKey.allShouldReportMetricKeys(instanceType);
+    for (MetricKey metricKey : metricKeys) {
+      SHOULD_REPORT_METRICS.putIfAbsent(
+          getMetricNameWithUniqueId(instanceType, metricKey.getName()),
+          metricKey.getMetricType());
```

Comment:
This method will cover most of the should report metrics which is `MetricKey` with `isClusterAggregated=true`. But it does not cover those with ufs tags like `Worker.BytesRead/WrittenPerUfs.UFS:<UFS_ADDRESS>.<worker_hostname>`. The code here doesn't know information about any of the Alluxio UFS. That's why I added the method `counterWithTags`  and `meterWithTags` which will add those full metric name to the should report metrics set.